sdk/dapp-kit/src/components/WalletProvider.tsx

```diff
@@ -60,6 +63,28 @@ export function WalletProvider({
 		connectionStatus: 'disconnected',
 	});
 
+	useWalletsChanged({
+		onWalletRegistered() {
```

Comment:
These probably should be a `useCallback` so that we don't re-run the underlying effect in `useWalletChanged` every render.

Reply:
Yep good call, I thought I needed to memoize the callbacks but wasn't 100% and hadn't reproduced an example to confirm it was needed. Updated!